### PR TITLE
reqResp cleanups

### PR DESCRIPTION
**Pull request by 3xtr4t3rr3str14l**
just a few things i found when making sure we are up to spec for reqResp:

-  remove unnecessary check in onBeaconBlocksByRange
-  remove unnecessary function params
-  clarifying error message in reqUtils->sendRequest

### CODE REVIEW

File: packages/lodestar/src/sync/reqResp/reqResp.ts [hunk 1]
```diff
@@ -97,11 +98,11 @@ export class BeaconReqRespHandler implements IReqRespHandler {
       case Method.Ping:
         return await this.onPing(request as ReqRespRequest<Ping>, peerId, sink);
       case Method.Metadata:
-        return await this.onMetadata(request as ReqRespRequest<null>, peerId, sink);
+        return await this.onMetadata(request as ReqRespRequest<null>, sink);
       case Method.BeaconBlocksByRange:
-        return await this.onBeaconBlocksByRange(request as ReqRespRequest<BeaconBlocksByRangeRequest>, peerId, sink);
+        return await this.onBeaconBlocksByRange(request as ReqRespRequest<BeaconBlocksByRangeRequest>, sink);
       case Method.BeaconBlocksByRoot:
-        return await this.onBeaconBlocksByRoot(request as ReqRespRequest<BeaconBlocksByRootRequest>, peerId, sink);
```
Comment (mpetrunic) [inline packages/lodestar/src/sync/reqResp/reqResp.ts]: I would keep peerId on those methods. We could/should track those requests by peerId so we can blacklist peer if requests too much blocks or something

File: packages/lodestar/src/sync/reqResp/reqResp.ts [hunk 2]
```diff
@@ -280,10 +280,6 @@ export class BeaconReqRespHandler implements IReqRespHandler {
       );
       return;
     }
-    if (request.body.count > 1000) {
-      this.logger.warn(`Request id ${request.id} asked for ${request.body.count} blocks, just return 1000 maximum`);
-      request.body.count = 1000;
-    }
```
Comment (mpetrunic) [inline packages/lodestar/src/sync/reqResp/reqResp.ts:286]: why did you remove this?
You could change this so that "1000" (maxRequestedBlocks) be an ReqResp option.

File: packages/lodestar/src/sync/utils/sync.ts [hunk 3]
```diff
@@ -226,10 +226,7 @@ export async function createStatus(chain: IBeaconChain): Promise<Status> {
 export async function syncPeersStatus(network: INetwork, status: Status): Promise<void> {
   await Promise.all(
     network.getPeers({connected: true, supportsProtocols: getStatusProtocols()}).map(async (peer) => {
-      try {
-        network.peerMetadata.setStatus(peer.id, await network.reqResp.status(peer.id, status));
-        // eslint-disable-next-line no-empty
-      } catch {}
+      network.peerMetadata.setStatus(peer.id, await network.reqResp.status(peer.id, status));
```
Comment (mpetrunic) [inline packages/lodestar/src/sync/utils/sync.ts]: we need try catch here otherwise it's unhandled rejection when `await network.reqResp.status(peer.id, status)` throws

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: packages/lodestar/src/network/reqresp/reqUtils.ts [hunk 4]
```diff
@@ -29,7 +29,7 @@ export async function sendRequest<T extends ResponseBody | ResponseBody[]>(
   signal?: AbortSignal
 ): Promise<T | null> {
   if (signal?.aborted) {
-    throw new Error("aborted");
+    throw new Error("sendRequest aborted");
```
Comment (dapplion) [inline packages/lodestar/src/network/reqresp/reqUtils.ts]: You can use `ErrorAborted` from lodestar-utils

File: packages/lodestar/src/sync/reqResp/reqResp.ts (hunk 2 above)
Comment (3xtr4t3rr3str14l) [reply packages/lodestar/src/sync/reqResp/reqResp.ts:286]: We have an identical check just below it with the correct number specified in the spec:
https://github.com/ChainSafe/lodestar/pull/1742/files#diff-7c5831b4dde5a221fa8a043078154dcd1d94033976ba1c25cc0e493ee3c7d156R284

**Review by mpetrunic — DISMISSED** [review]

File: packages/lodestar/src/network/reqresp/reqUtils.ts [hunk 5]
```diff
@@ -29,7 +29,7 @@ export async function sendRequest<T extends ResponseBody | ResponseBody[]>(
   signal?: AbortSignal
 ): Promise<T | null> {
   if (signal?.aborted) {
-    throw new Error("aborted");
+    throw new ErrorAborted("sendRequest aborted");
```
Comment (dapplion) [inline packages/lodestar/src/network/reqresp/reqUtils.ts]: This would print `Aborted sendRequest aborted`. Either modify the `ErrorAborted` constructor or this message

**Review by dapplion — DISMISSED** [review]

**Review by wemeetagain — APPROVED** [review]
